### DISCUSSION TITLE
Export limit reached

### DISCUSSION THREAD

**Discussion by wickzter** [wills106/homeassistant-solax-modbus#490] [Q&A]
Hi All, 

As many have stated awesome project, thank you...

I have a 5kW Solax X1 Gen 4, with 5.6kW peak panels, unfortunately the grid have imposed an export limit of 3.68kW, is there a way I can tell through the sensors if this has been reached, so I could active something to use the excess?

Thanks in advance

**Comment by wills106** [comment]
Are you in the UK by any chance on a G98 Installation?

**Reply by wickzter** [reply]
I am in the UK, not sure on the G98, the installers applied for a fast track, the the DNO (National Grid) came back and said they needed to do an investigation, then after delaying for 2 months said yes but rate limit of 3.68kW.  Bit of a pain, and in reality I can't see that I will be exporting more than that regularly, but I have definitely seen the limit hit the last couple of days, once I turn on something power hungry I can see the PV jump up.

**Reply by wills106** [reply]
You can set your export limit to 3680w then or there abouts.

It's not a quantity your limited to, but 3.68kw at any time.

So you can export 30kWh in a day, as long as you only export 3.68kW at any given time.

**Reply by wickzter** [reply]
Yes, this is how it is set, however there are times when I am hitting that limit.  In theory I could be clipping from 5.6kW to 3.68kW, so wasting almost 2kW opportunity.  I know there is a house load of about 300W in my case.  But I could switch on my 2kW immersion heater and use the excess.  Hence my original question, is there a way to know when I have hit the export limit, and how much I am clipping by?

I hope this makes sense and thanks for your replies.

**Reply by wills106** [reply]
I misread thinking you wanted to just limit your export to comply.

You could have an automation based off Grid Export sensor. If exporting 3600w turn on immersion.
If grid export 200w or less turn off immersion.

Then you won't discharge your battery.

**Comment by wickzter** [comment]
Thanks Will, that would work.  Ideally I would want to understand how much I was clipping by and then activate the immersion at a given value, so I am still exporting some.  What I have noticed is that when I hit the export limit the value on the PV drops accordingly to match load + export.  If it's really sunny and I put the kettle in I see the load increase and the PV increase at the same time.

Ultimately I get paid more per kWh of electricity for export than the gas costs me to otherwise heat the water.  I'm really only musing at this time as I don't have a smart way of activating the immersion (yet).

**Reply by wills106** [reply]
Be careful with those WiFi switches. Ideally you want to trigger a contactor to switch the load. But put a delay so it's not switching on / off every second etc.

Other option is to build / by an actual solar diverter and configure it so it only kicks in around 3500w export and then it always regulates the diversion into your hot water.

**Reply by wickzter** [reply]
Originally I was thinking about getting an Eddie or Solar iBoost, but Octopus Agile or Flux kind of make that redundant, except in my case where I am clipping and could use that excess.

I had considered a Shelly 1 PM, but I'm nervous about switching that load even though the Shelly is rated at 16A, various horror stories abut them getting very hot.  And worse Sonoff devices melting.
